documentation/modules/cruise-control/con-cruise-control-overview.adoc

```diff
@@ -2,54 +2,508 @@
+* *Soft goals* are objectives that are prioritized during optimization as much as possible, without preventing a proposal from being created if all hard goals are satisfied.
+* *Default goals* refer to the goals used by default when generating proposals. 
+They match the main goals unless specifically set by the user.
+* *Proposal-specific goals* are a subset of default goals configured for specific proposals.
+
+Configure optimization goals in the `Kafka` and `KafkaRebalance` custom resources.
+
+* `Kafka` resource for main, hard, and default goals.
+** Main goals: `Kafka.spec.cruiseControl.config.goals`
+** Hard goals: `Kafka.spec.cruiseControl.config.hard.goals`
+** Default goals: `Kafka.spec.cruiseControl.config.default.goals`
+* `KafkaRebalance` resource for proposal-specific goals.
+** Proposal-specific goals: `KafkaRebalance.spec.goals`
+
+=== Main goals
+
+Main goals are predefined and available to all users in Cruise Control. 
+Goals not listed as main goals cannot be used in Cruise Control operations.
+Some main goals are preset as hard goals.
+
+To simplify configuration, use the inherited main goals unless you need to exclude specific goals from `KafkaRebalance` resources. 
+You can adjust the priority order in the default optimization goals configuration.
+
+Configure main goals in `Kafka.spec.cruiseControl.config.goals`:
+
+* To accept inherited main goals, omit the `goals` property.
+* To modify main goals, specify the goals in descending priority order in the `goals` property.
+
+=== Hard and soft goals
+
+Hard goals are mandatory and must be satisfied for optimization proposals to be generated. 
+Soft goals are best-effort objectives that Cruise Control tries to meet after all hard goals are satisfied.
+The classification of hard and soft goals is fixed in Cruise Control code and cannot be changed.
+
+Cruise Control first prioritizes satisfying hard goals, and then addresses soft goals in the order they are listed.
+A proposal meeting all hard goals is valid, even if it violates some soft goals. 
+
+For example, a soft goal might be to evenly distribute a topic's replicas. 
+Cruise Control will ignore this if it conflicts with hard goals.
+
+Configure hard goals in your Cruise Control deployment using `Kafka.spec.cruiseControl.config.hard.goals`:
+
+* To enforce all hard goals, omit the `hard.goals` property.
+* To specify hard goals, list them in `hard.goals`.
+* To exclude a hard goal, ensure it's not in either `default.goals` or `hard.goals`.
+
+Increasing the number of configured hard goals will reduce the likelihood of Cruise Control generating optimization proposals.
+
+=== Default goals
+
+Cruise Control uses default goals to generate an optimization proposal. 
+You can override default goals by setting proposal-specific optimization goals in a `KafkaRebalance` resource.
+
+If `default.goals` is not specified in the Cruise Control deployment configuration, main goals are used as default goals. 
```

Comment:
```suggestion
If `default.goals` is not specified in the Cruise Control deployment configuration, Strimzi will configure the `default.goals` to the list of goals specified in `main.goals`. 
```

Reply:
I think we mean `config.goals` not `main.goals`?